whipper/common/encode.py

```diff
@@ -89,3 +90,71 @@ def _tag(self):
         w.save()
 
         self.stop()
+
+
+class EmbedPictureTask(task.Task):
+    description = 'Embed picture to FLAC'
+
+    def __init__(self, track_path, coverart_path):
+        self.track_path = track_path
+        self.coverart_path = coverart_path
+
+    def start(self, runner):
+        task.Task.start(self, runner)
+        self.schedule(0.0, self._embed_picture)
+
+    def _make_flac_picture(self, coverart_filename):
+        """
+        Given a path to a jpg/png file, return a FLAC picture for embedding.
+
+        The embedding will be performed using the mutagen module.
+
+        :param coverart_filename: path to cover art image file
+        :type coverart_filename: str
+        :returns: a valid FLAC picture for embedding
+        :rtype: mutagen.flac.Picture or None
+        """
+        if not coverart_filename:
+            return
+
+        from PIL import Image
+
+        im = Image.open(coverart_filename)
+        if im.format == 'JPEG':
+            mime = 'image/jpeg'
+        elif im.format == 'PNG':
+            mime = 'image/png'
+        else:
+            logger.warning("no cover art will be added because the fetched "
+                           "image format is unsupported")
+            return               # we only support png and jpeg
+
+        pic = Picture()
+        with open(coverart_filename, 'rb') as f:
+            pic.data = f.read()
+
+        pic.type = PictureType.COVER_FRONT
+        pic.mime = mime
```

Comment:
Why are we saving MIME type here?

Reply:
This is originally from a comment in [#50](https://github.com/whipper-team/whipper/issues/50#issuecomment-422144811).

Reply:
> Why are we saving MIME type here?

From [mutagen's docs](https://mutagen.readthedocs.io/en/latest/api/flac.html#mutagen.Picture.data):
> To create a picture from file (in order to add to a FLAC file), instantiate this object without passing anything to the constructor and then set the properties manually:
> 
> ```python
> p = Picture()
> 
> with open("Folder.jpg", "rb") as f:
>     pic.data = f.read()
> 
> pic.type = id3.PictureType.COVER_FRONT
> pic.mime = u"image/jpeg"
> pic.width = 500
> pic.height = 500
> pic.depth = 16 # color depth
> ```